include/ginkgo/core/base/executor.hpp

```diff
@@ -50,6 +50,55 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/synthesizer/containers.hpp>
 
 
+namespace gko {
+
+
+/**
+ * Specify the mode of allocation for CUDA/HIP GPUs.
+ *
+ * `device` allocates memory on the device and Unified Memory model is not used.
+ *
+ * `unified_global` allocates memory on the device, but is accessible by the
+ * host through the Unified memory model.
+ *
+ * `unified_host` allocates memory on the
+ * host and it is not available on devices which do not have concurrent acesses
+ * switched on, but this access can be explictly switched on, when necessary.
+ */
+enum class allocation_mode { device, unified_global, unified_host };
+
+
+#ifdef NDEBUG
```

Comment:
Do you think we should use another cmake variable to control it? or user change this file to change the behavior

Reply:
No, the user controls it by creating a executor capable of allocating on unified memory. I can add a setter, but I think that is confusing and error-prone. We can anyway create multiple executors on one CUDA device

Reply:
I see